cpp/modmesh/buffer/ConcreteBuffer.hpp

```diff
@@ -117,8 +118,8 @@ struct ConcreteBufferDataDeleter
 /**
  * Untyped and unresizeable memory buffer for contiguous data storage.
  */
-class ConcreteBuffer
-    : public std::enable_shared_from_this<ConcreteBuffer>
+class ConcreteBuffer : public BufferBase<ConcreteBuffer>
```

Comment:
`std::enable_shared_from_this` should be the first parent.

Format:

```cpp
class ConcreteBuffer
     : public BufferBase<ConcreteBuffer>
     , public std::enable_shared_from_this<ConcreteBuffer>
```

Reply:
fixed.